src/ArborX_LinearBVH.hpp

```diff
@@ -55,27 +55,17 @@ class BoundingVolumeHierarchy
   KOKKOS_FUNCTION
   bounding_volume_type bounds() const noexcept { return _bounds; }
 
-  template <typename ExecutionSpace, typename Predicates,
-            typename CallbackOrView, typename View, typename... Args>
+  template <typename ExecutionSpace, typename Predicates, typename... Args>
   void query(ExecutionSpace const &space, Predicates const &predicates,
-             CallbackOrView &&callback_or_view, View &&view,
              Args &&... args) const
```

Comment:
Do you want a single `query` function for everything? I would probably encourage being explicit in the signature of the third argument and having all three signatures explicitly visible.